cmd/registry/controller/parser.go

```diff
@@ -84,17 +88,24 @@ func resourceNameFromDependency(
 	//    returns projects/demo/locations/global/apis/petstore/versions/1.0.0/specs/openapi.yaml/artifacts/custom-artifact
 
 	// Replace apis/- pattern with the corresponding name and so on.
-	apiPattern := regexp.MustCompile(`.*(/apis/)-`)
-	resourceName := apiPattern.ReplaceAllString(resourcePattern, fmt.Sprintf("%s", dependency.GetApi()))
+	// dependency.GetApi() returns the full api name projects/demo/locations/global/apis/petstore
+	// We use stringsSplit()[-1] to extract only the API name
+	// apiPattern := regexp.MustCompile(`/apis/-`)
+	apiName := strings.Split(dependency.GetApi(), "/")
+	resourceName := strings.Replace(resourcePattern, "/apis/-",
+		fmt.Sprintf("/apis/%s", apiName[len(apiName)-1]), -1)
```

Comment:
Nit: `strings.ReplaceAll` seems like a better function to use here so you don't have to pass the `-1` each time.

Reply:
Done.